client/src/components/modals/createDataset/DataFileDataSourceSettings.jsx

```diff
@@ -35,17 +42,23 @@ export default class DataFileDataSourceSettings extends Component {
     this.uploadFile(evt.dataTransfer.files[0]);
   }
 
+  handleProgress(percentage) {
+    this.setState({ uploadProgressPercentage: parseFloat(percentage) });
+  }
+
   uploadFile(file) {
     const onChange = this.props.onChange;
+    const handleProgress = this.handleProgress;
     const upload = new tus.Upload(file, {
       headers: { Authorization: `Bearer ${keycloak.token}` },
       endpoint: '/api/files',
       onError(error) {
         console.error(`Failed because: ${error}`);
+        handleProgress(null);
```

Comment:
In `handleProgress` we `parseFloat` the `percentage` argument and `parseFloat(null)` evaluates to `NaN` which I don't think is desired here, especially since we explicitly check for `null` in `getProgressBarBool`.